.github/workflows/release.yml

```diff
@@ -0,0 +1,189 @@
+
+      - name: Release AppImage
+        if: startsWith(matrix.config.os, 'ubuntu')
+        uses: softprops/action-gh-release@v1
+        with:
+          draft: true
+          prerelease: ${{ steps.get_version.outputs.ISBETA }}
+          files: build/cpeditor-${{ steps.get_version.outputs.VERSION }}-linux-x86_64${{ matrix.config.portable-name }}.AppImage
+          name: CP Editor ${{ steps.get_version.outputs.VERSION }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Package full Source
+        if: startsWith(matrix.config.os, 'ubuntu') && matrix.config.portable-option == 'Off'
+        run: |
+          wget https://raw.githubusercontent.com/Kentzo/git-archive-all/master/git_archive_all.py
+          python3 git_archive_all.py cpeditor-${{ steps.get_version.outputs.VERSION }}-full-source.tar.gz
+
+      - name: Release full source
+        if: startsWith(matrix.config.os, 'ubuntu') && matrix.config.portable-option == 'Off'
+        uses: softprops/action-gh-release@v1
+        with:
+          draft: true
+          prerelease: ${{ steps.get_version.outputs.ISBETA }}
+          files: cpeditor-${{ steps.get_version.outputs.VERSION }}-full-source.tar.gz
+          name: CP Editor ${{ steps.get_version.outputs.VERSION }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Pack to dmg on Mac
+        if: startsWith(matrix.config.os, 'macos')
+        run: |
+          cd build
+          ../../Qt/*/*/bin/macdeployqt cpeditor.app
+          cp ../.ci/mac/cpeditor.icns cpeditor.app/Contents/Resources
+          hdiutil create -volname cpeditor-${{ steps.get_version.outputs.VERSION }}-x64 -srcfolder cpeditor.app -ov -format UDZO cpeditor-${{ steps.get_version.outputs.VERSION }}-macos-x64${{ matrix.config.portable-name }}.dmg
+
+      - name: Release on Mac
+        if: startsWith(matrix.config.os, 'macos')
+        uses: softprops/action-gh-release@v1
+        with:
+          draft: true
+          prerelease: ${{ steps.get_version.outputs.ISBETA }}
+          files: build/cpeditor-${{ steps.get_version.outputs.VERSION }}-macos-x64${{ matrix.config.portable-name }}.dmg
+          name: CP Editor ${{ steps.get_version.outputs.VERSION }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Gather dependencies on Windows
+        if: startsWith(matrix.config.os, 'windows')
+        shell: bash
+        run: |
+          mkdir build/out
+          mv build/Release/cpeditor.exe build/out
+          ../Qt/*/*/bin/windeployqt.exe build/out/cpeditor.exe --no-translations
+
+      - name: Pack to installer on Windows
+        if: startsWith(matrix.config.os, 'windows') && matrix.config.portable-option == 'Off'
+        shell: bash
+        run: |
+          mkdir build/out/Redist
+          cp "C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Redist/MSVC/$VC_REDIST_VERSION/vc_redist.x64.exe" build/out/Redist
```

Comment:
Suppose this cp has failed, We will have a partial release because some binaries will be released like dmg and appimage, but this will not, what will be done then? delete the complete tag and release again? A hard work.

Reply:
> Suppose this cp has failed, We will have a partial release because some binaries will be released like dmg and appimage, but this will not, what will be done then? delete the complete tag and release again? A hard work.

I spent more than one minute to understand what you mean, and I think this is much better:

If this `cp` fails, we will have a release in which there are only the `dmg`s and `AppImage`s but not the `exe`. When that happens, what should we do? If we delete the tag and release again, that will be hard work.